// react-code-blocks/build/esm/components/CopyBlock.js
import _extends from "@babel/runtime/helpers/extends";
import _slicedToArray from "@babel/runtime/helpers/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React, { useState } from 'react';
import Code from './Code';
import CodeBlock from './CodeBlock';
import Copy from './Copy';
import Clipboard from 'clipboard';
import styled from 'styled-components';
var Button =
/*#__PURE__*/
styled.button.withConfig({
  displayName: "CopyBlock__Button",
  componentId: "sc-109lwr7-0"
})(["position:absolute;top:0;right:0;margin-right:0.25rem;display:flex;flex-wrap:wrap;justify-content:center;align-items:center;background:", ";margin-top:0.15rem;border-radius:0.25rem;max-height:2rem;max-width:2rem;padding:0.5rem;&:hover{opacity:", ";}&:focus{outline:none;opacity:1;}.icon{width:1rem;height:1rem;}"], function (p) {
  return p.theme.backgroundColor;
}, function (p) {
  return p.copied ? 1 : 0.5;
});
var Snippet =
/*#__PURE__*/
styled.div.withConfig({
  displayName: "CopyBlock__Snippet",
  componentId: "sc-109lwr7-1"
})(["display:flex;flex-wrap:wrap;position:relative;background:", ";border-radius:0.25rem;padding:", ";"], function (p) {
  return p.theme.backgroundColor;
}, function (p) {
  return p.codeBlock ? "0.25rem 0.5rem 0.25rem 0.25rem" : "0.25rem";
});

var uniqueId = require("lodash.uniqueid");

export default function (_ref) {
  var theme = _ref.theme,
      text = _ref.text,
      codeBlock = _ref.codeBlock,
      rest = _objectWithoutProperties(_ref, ["theme", "text", "codeBlock"]);

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      copied = _useState2[0],
      toggleCopy = _useState2[1];

  var uid = uniqueId("copy_");

  if (typeof document !== "undefined") {
    var clip = new Clipboard("#".concat(uid));
    clip.on("success", function () {
      toggleCopy(true);
      return setTimeout(function () {
        return toggleCopy(false);
      }, 1000);
    });
    clip.on("error", function (err) {
      return console.error(err);
    });
  }

  return React.createElement(Snippet, _extends({
    codeBlock: codeBlock
  }, {
    theme: theme
  }), codeBlock ? React.createElement(CodeBlock, _extends({
    text: text,
    theme: theme
  }, rest)) : React.createElement(Code, _extends({
    text: text,
    theme: theme
  }, rest)), React.createElement(Button, _extends({
    id: uid
  }, {
    theme: theme,
    copied: copied
  }, {
    "data-clipboard-text": text,
    disabled: copied
  }), React.createElement(Copy, {
    color: copied ? theme.stringColor : theme.textColor,
    copied: copied,
    className: "icon"
  })));
}
